### PR TITLE
Add integration test for sxp elements

### PR DESCRIPTION
**Pull request by gustavolimav**
https://issues.liferay.com/browse/LPS-149140
https://issues.liferay.com/browse/LPS-149138
https://issues.liferay.com/browse/LPS-149142
https://issues.liferay.com/browse/LPS-149143
https://issues.liferay.com/browse/LPS-149135

### CODE REVIEW

**Comment by gustavolimav** [comment]
ci:relevant

**Comment by gustavolimav** [comment]
ci:stable

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:stable

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:stable

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:stable

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:stable

**Comment by gustavolimav** [comment]
ci:test:stable

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:stable

File: modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java [hunk 1]
```diff
@@ -1156,6 +1156,89 @@ public void testLimitSearchToContentsCreatedWithinAPeriodOfTime()
...
+					"slop", 0
+				).put(
+					"type", "best_fields"
+				).build()
+			},
+			new String[] {"Text Match Over Multiple Fields"});
+
+		_keywords = "Article";
+
+		_assertSearch("[Article 1.2, Article 1.1, Article 1.0]");
```
Comment (joshuacords) [inline modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java:1200]: Looks potentially flaky, don't we want to use _assertSearchIgnoreRelevance here?

File: modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java [hunk 2]
```diff
@@ -1240,6 +1240,79 @@ public void testLimitSearchToPDFFiles() throws Exception {
...
+					"type", "best_fields"
+				).build(),
+				null
+			},
+			new String[] {
+				"Text Match Over Multiple Fields",
+				"Limit Search to Published Contents"
+			});
+
+		_assertSearch("[]");
```
Comment (joshuacords) [inline modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java:1325]: Here and a few other places, we want to make sure that the filtering doesn't just remove results but also still allows some through. Add a published JournalArticle to this test that we would still expect to be returned in this last case.

File: modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java [hunk 3]
```diff
@@ -1550,6 +1550,86 @@ public void testPhraseMatch() throws Exception {
...
+				).put(
+					"type", "best_fields"
+				).build(),
+				null
+			},
+			new String[] {
+				"Text Match Over Multiple Fields", "Scheduling Aware"
+			});
+
+		_assertSearch("[]");
```
Comment (joshuacords) [inline modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java:1528]: Here and a few other places, we want to make sure that the filtering doesn't just remove results but also still allows some through. Add an unscheduled JournalArticle to this test that we would still expect to be returned in this last case.

File: modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java [hunk 4]
```diff
@@ -1712,6 +1717,80 @@ public void testSearch() throws Exception {
...
+				).put(
+					"slop", 0
+				).put(
+					"type", "best_fields"
+				).build(),
+				null
+			},
+			new String[] {"Text Match Over Multiple Fields", "Staging Aware"});
+
+		_assertSearch("[]");
```
Comment (joshuacords) [inline modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java:1629]: Here and a few other places, we want to make sure that the filtering doesn't just remove results but also still allows some through. Add a unstaged JournalArticle to this test that we would still expect to be returned in this last case.

File: modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java [hunk 5]
```diff
@@ -959,6 +961,56 @@ public void testHideByExactTermMatch() throws Exception {
...
+
+		_assertSearch("[Comment]");
+
+		_updateElementInstancesJSON(
+			new Object[] {
+				_getDefaultValuesForTextMatchOverMultipleFields(), null
+			},
+			new String[] {"Text Match Over Multiple Fields", "Hide Comments"});
+
+		_assertSearch("[]");
```
Comment (joshuacords) [inline modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java:1002]: Here and a few other places, we want to make sure that the filtering doesn't just remove results but also still allows some through. Add a JournalArticle or something like this to this test (and add that type to the searchable asset types) that we would still expect to be returned in this last case.

**Comment by joshuacords** [comment]
Hi @gustavolimav,
Finally got around to reviewing this. I really like the refactoring commits, just a couple of comments for changes to strengthen the tests for fuller coverage.

File: modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java (hunk 1 above)
Comment (gustavolimav) [reply modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java:1200]: Yeah make sense

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:stable

**Comment by joshuacords** [comment]
Thanks for the changes @gustavolimav, looks good to me. I'll forward these once the test finish.

**Comment by joshuacords** [comment]
In the future - to preserve the change history, just open a new PR instead of reopening the same one so it's easier for reviewers to follow what has changed and what was commented on.

**Comment by joshuacords** [comment]
ci:forward

**Pull request closed without merging**